Dev

uriwin opened the pull request.

uri-win committed:
added prject code

uri-win committed:
Print equipment details and summary for each team member

uri-win committed:
changed to switch instead of if in TechnologyEquipmentFactory.java UniformEquipmentFactory.java

equipmentType() now is a function in the abstract classes.
changed name from Harmonit to Hermonit.

uri-win committed:
added enum for UniformEquipmentTypes.java TechnologyEquipmentTypes.java.

use enum instead of string when creating objects in UniformEquipmentFactory.java TechnologyEquipmentFactory.java

uri-win committed:
changed equipmentType() to getEquipmentType().

instead of one function that print the TeamMember Equipment details and summary, created two functions:
getMemberEquipmentDetails, getMemberEquipmentSummary.

uri-win committed:
functions in EquipmentStatus.java are not static anymore.

Removed generic <T> (no need), and now the functions get input of HashMap<String, List<Equipment>> instead.
TeamMember have memberEquipments instead of memberUniformEquipment and memberTechnologyEquipment for OCP.

uri-win committed:
added ITeamMember and AbstractTeamMember for DIP.

rename TechnologyEquipment.java to AbstractTechnologyEquipment because of conventions (same goes for AbstractUniformEquipment and IEquipment).

uri-win committed:
changed vars names for conventions.

uri-win committed:
seperate EquipmentSummary and EquipmentDetails functions.

created constructors for the abstract classes.

uri-win committed:
fixing CR.

uri-win committed:
getTeamSummary is now more OCP.

uri-win committed:
chamged location of summarizer

uri-win committed:
CR changes.

noam-toledano commented on src/main/java/equipment/describer/TeamEquipmentDescriber.java:
```diff
@@ -0,0 +1,22 @@
+import java.util.Map;
+
+public class TeamEquipmentDescriber {
+    public String getTeamEquipmentDescription(Map<ITeamMember, List<AbstractEquipment>> teamEquipments) {
```
if you only use the keys of the map, why not just get a list of team members and use their `getEquipment` method?

noam-toledano commented on src/main/java/equipment/technology/TechnologyEquipmentFactory.java:
```diff
@@ -0,0 +1,20 @@
+            default:
+                System.out.println("The equipment type: " + technologyEquipmentType.toString() + " does not exists");
+        }
+        throw new IllegalArgumentException("Invalid Equipment type");
```
put this section in the `default` case- it makes the code's flow more readable at a quick glance

noam-toledano commented on src/main/java/summarizer/team/GetTeamSummary.java:
```diff
@@ -0,0 +1,21 @@
+import java.util.List;
+import java.util.Map;
+
+public class GetTeamSummary {
```
class names should be nouns, method names should be verbs

noam-toledano commented on src/main/java/summarizer/team/GetTeamSummary.java:
```diff
@@ -0,0 +1,21 @@
+    public String getTeamSummary(TeamSummarizerType teamSummarizerType, Map<ITeamMember, List<AbstractEquipment>> teamEquipments){
+        switch (teamSummarizerType){
+            case NormalTeamEquipmentSummarizer:
+                return new NormalTeamEquipmentSummarizer().getTeamSummary(teamEquipments);
```
SRP- why not return the summarizer and use it elsewhere?

noam-toledano commented on src/main/java/summarizer/team/NormalTeamEquipmentSummarizer.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Override
+    public String getTeamSummary(Map<ITeamMember, List<AbstractEquipment>> teamEquipments) {
```
the extended summarizer is supposed to output each individual equipment's full description- not a summary of amount and type

noam-toledano commented on src/main/java/summarizer/team/NormalTeamEquipmentSummarizer.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Override
+    public String getTeamSummary(Map<ITeamMember, List<AbstractEquipment>> teamEquipments) {
+        StringBuilder teamEquipmentSummarizer = new StringBuilder();
```
if the `StringBuilder` doesn't summarize, it can't be called a `Summarizer`

noam-toledano commented on src/main/java/summarizer/team/NormalTeamEquipmentSummarizer.java:
```diff
@@ -0,0 +1,37 @@
+        StringBuilder teamEquipmentSummarizer = new StringBuilder();
+        SummarizerUtils summarizerUtils = new SummarizerUtils();
+        for (ITeamMember teamMember: teamEquipments.keySet()) {
+            teamEquipmentSummarizer.append(teamMember.getName()).append(" from team ").append(teamMember.getTeamName())
```
add newlines in the code between these long `append` chains for added readability

uri-win committed:
when building string new lines now have separate lines so the code is more readable.

functions and vars Names are more indicative.
ExtendedTeamEquipmentSummarizer returns each team member equipment description and appearances count.

uri-win committed:
same as last commit changes
